TESTING.md

```diff
@@ -24,15 +24,28 @@ before writing new integration tests, which are slower to run and are not as eas
 Before running tests, make sure your Flutter SDK matches the version that will be used on
 the CI. To update your local flutter version, run:
 
+```shell
+flutter channel main
+flutter upgrade
```

Comment:
for development on the DevTools repo we assume flutter is being vendored from git. We have a repo management tool to manage flutter versioning, but looks like the command was just stale.

This should be replaced with `devtools_tool update-flutter-sdk --update-on-path`

Reply:
This command doesn't work unless you do `(cd tool; dart pub get)` before running it, and it also ignores the flutter on the path, and tries to make a whole new clone of the repo in the `tool` directory.

(I tested this by running `git clean -dxf` and then trying these instructions.)

Reply:
Ahh, yes the linked instructions do say to do the `dart pub get`, so I guess that doesn't need to be mentioned here, but it still is ignoring the flutter in my path and cloning a new repo.

Reply:
It does appear to be finding the flutter from the right place, however:

```
Running devtools_tool using the Dart SDK from /usr/local/google/home/gspencer/code/flutter/bin/cache/dart-sdk/bin/dart to create the Flutter SDK in tool/flutter-sdk.
Using Flutter SDK from /usr/local/google/home/gspencer/code/flutter
Updating to Flutter version from cache: 8925e1ffdfe880b06a8bc5877f017083d6652f5b
Cloning Flutter into /usr/local/google/home/gspencer/code/devtools/tool/flutter-sdk
/usr/local/google/home/gspencer/code/devtools/tool > git clone https://github.com/flutter/flutter flutter-sdk
Cloning into 'flutter-sdk'...
```

Reply:
Yes the new flutter clone is intentional. DevTools uses a pinned version of Flutter so this is to make sure we are testing and building with the same flutter version DevTools will be shipped with.